### PR TITLE
Add option to hide dock icon

### PR DESCRIPTION
**Pull request by dcalhoun**
Allow users to optionally hide the Hyper icon from the dock and application switcher.

### Before this can be merged...
- [x] Merge soutar/hyperterm-register-shortcut#5.
- [x] Update hyperterm-summon's dependency range for hypterm-register-shortcut.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,58 +1,9 @@
...
-  if (focusedWindows.length > 0) {
-    hideWindows(windows, app);
-  } else {
-    showWindows(windows, app);
-  }
-}
-
-function addWindow (window) {
-  windowSet.add(window);
-  window.on('focus', () => {
```
Comment (dcalhoun) [inline index.js]: @soutar I removed this listener when reformatting the code to support the changes I needed. It didn't appear this listener was necessary. Will you double check that for me? I may have overlooked something.

Comment (soutar) [inline index.js]: I *think* the purpose of this listener was to ensure that the order of the windowSet represented the visual order of the windows, with the most recently focused on top and the least recently focused on the bottom. If you have multiple windows open and you hide them, is the order still correct after you summon with this change?

Comment (dcalhoun) [inline index.js]: Nope, I did notice that behavior. Good catch. I'll bring this block of code back to fix that.

Comment (soutar) [inline index.js]: Sweet. It's been a while so thanks for calling this out, I might have missed it otherwise

**Comment by soutar** [comment]
Would you mind adding a `.travis.yml` file [like the one on hyperterm-register-shortcut](https://github.com/soutar/hyperterm-register-shortcut/blob/master/.travis.yml)? I've just turned on Travis CI so it should pick it up and start running the tests for us

**Comment by soutar** [comment]
I'm not in a position to test this properly at the moment but it looks good. I'm happy to merge this if it's working for you 👍

**Comment by dcalhoun** [comment]
@soutar yeah, I have it aliased locally at the moment. It's working as expected for me.

**Pull request merged**